### PR TITLE
fix(dialog): do not render overlay in fullscreen size

### PR DESCRIPTION
**Pull request by Powerplex**
### Description, Motivation and Context

The fadeOut applied to the overlay of the dialog when it's closing was looking weird on `fullscreen` size.

I introduce a context so that the `Dialog.Overlay` is aware of the size passed to `Dialog.Content` and is never rendered in `fullscreen` size.

### Types of changes

- [x] 🪲 Bug fix (non-breaking change which fixes an issue)

### Screenshots - Animations

https://github.com/adevinta/spark/assets/2033710/27e12119-e7f8-4360-b3f6-82b7b6bcb786

### CODE REVIEW

File: packages/components/dialog/src/DialogContent.tsx [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
-        {...rest}
-      >
-        {children}
-      </RadixDialog.Content>
-    </div>
-  )
+  ): ReactElement => {
+    const { setIsFullScreen } = useDialog()
+
+    useLayoutEffect(() => {
```
Comment (acd02) [inline packages/components/dialog/src/DialogContent.tsx:20]:
```suggestion
    useEffect(() => {
```

Comment (Powerplex) [reply packages/components/dialog/src/DialogContent.tsx:20]: Why ? I prefer `useLayoutEffect` it's good for this case: https://react.dev/reference/react/useLayoutEffect

Comment (acd02) [reply packages/components/dialog/src/DialogContent.tsx:20]: > Usage: Measuring layout before the browser repaints the screen
That is not the case here, you don't read values from an HTML element 🤷

Comment (Powerplex) [reply packages/components/dialog/src/DialogContent.tsx:20]: No by I need this value updated in the state before the component is rendered, not after.

**Review by kikoruiz — APPROVED** [review]

File: packages/components/dialog/src/DialogContent.tsx (hunk 1 above)
Comment (acd02) [reply packages/components/dialog/src/DialogContent.tsx:20]: Yes, we can discuss it this morning 👍

**Review by acd02 — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Pull request merged**